homeworks/05-redux-basic/002-basic-redux/src/redux/reducer.ts

```diff
@@ -0,0 +1,16 @@
+import { State, Action, ActionType } from "./types"
+
+export function reducer(state: State = 1000.0, action: Action<number>): State {
+  switch (action.type) {
+    case ActionType.UpdateBalance:
+      return action.payload
+    case ActionType.Credit:
+      return state - action.payload
+    case ActionType.Debit:
+      return state + action.payload
+    case ActionType.SubstractPercentage:
+      return state * (1 - action.payload / 100)
```

Comment:
The only note would be that we can make balance more clear. Because otherwise we get number like this `810.9228690634445` 🙂

Reply:
Thank you, Anton :)
That's important note